### PR TITLE
ext-imagick: support version string from ImageMagick 6.x

### PR DESCRIPTION
**Pull request by glensc**
Fixes: #8055 

To test, invoke:
```
$ ./bin/composer show -p | grep -i imagick
ext-imagick          3.4.3     The imagick PHP extension
lib-imagick          7.0.8.34  The imagick PHP library
```

refs:
- https://github.com/composer/composer/pull/7762#issuecomment-476106645
- https://github.com/composer/composer/pull/7762

### CODE REVIEW

File: src/Composer/Repository/PlatformRepository.php [hunk 1]
```diff
@@ -166,8 +166,14 @@ protected function initialize()
                 case 'imagick':
                     $imagick = new \Imagick();
                     $imageMagickVersion = $imagick->getVersion();
-                    preg_match('/^ImageMagick ([\d.]+)-(\d+)/', $imageMagickVersion['versionString'], $matches);
-                    $prettyVersion = "{$matches[1]}.{$matches[2]}";
+                    // 6.x: ImageMagick 6.2.9 08/24/06 Q16 http://www.imagemagick.org
+                    // 7.x: ImageMagick 7.0.8-34 Q16 x86_64 2019-03-23 https://imagemagick.org
+                    preg_match('/^ImageMagick ([\d.]+)(?:-(\d+))/', $imageMagickVersion['versionString'], $matches);
```
Comment (alcohol) [inline src/Composer/Repository/PlatformRepository.php]: Should it not be `'/^ImageMagick ([\d.]+)(?:-(\d+))?/'` (note the extra `?` on the non-capturing group)

**Comment by alcohol** [comment]
Should probably target the `1.8` branch since it is a bugfix.

File: src/Composer/Repository/PlatformRepository.php (hunk 1 above)
Comment (glensc) [reply src/Composer/Repository/PlatformRepository.php]: yes, indeed. thanks for the review. I have no excuse why I did not actually test this.

Comment (alcohol) [reply src/Composer/Repository/PlatformRepository.php]: No worries, it is Monday. That's always a great excuse ;-)

**Comment by Seldaek** [comment]
Thanks, merged in 1.8/master as 12e683e2e

**Pull request closed without merging**